src/Models/ModelInstantiator.php

```diff
@@ -252,7 +255,7 @@ public function dataObjectToArray(
             }
 
             // This will cleanse any private data from the data object if necessary.
-            if ($data_obj instanceof IPrivacyProtectedDataObject) {
+            if ($data_obj instanceof IPrivacyProtectedDataObject && $cleanse_privacy === true) {
```

Comment:
I was mildly curious as to whether `instanceof` performance differs significantly from `=== true` and if changing the order of the conditionals would have any impact on performance, so I ran some benchmarks.

My understanding is that PHP uses string comparison to determine `instanceof`, and I observed that longer class names did indeed take longer to evaluate.

In testing, `instance of PDGA\DataObjects\Interfaces\IPrivacyProtectedDataObject` took more than 2x the time to evaluate versus `$someVar === true`. So I figured changing the order of the conditionals would result in better performance. However, when I benchmarked the two versions, I didn't see much difference. I also tried loose comparison with the boolean and didn't see much difference there, either.

<img width="642" alt="image" src="https://github.com/user-attachments/assets/3c6745ea-2392-4cc0-b931-20ac710b393a">